app/models/specific_tube_rack_creation.rb

```diff
@@ -0,0 +1,436 @@
+  end
+
+  # Creates racked tubes based on the provided attributes and associates them with a new tube rack.
+  #
+  # This method iterates over the array of racked tube attributes provided in the rack_attributes
+  # hash.
+  # For each set of tube attributes, it calls the create_racked_tube method to create and associate
+  # the tube with the new tube rack.
+  #
+  # @param [Array<Hash>] racked_tubes An array of hashes, each containing attributes for a racked tube.
+  # @param [TubeRack] new_tube_rack The new tube rack object to which the tubes will be associated.
+  #
+  # @raise [StandardError] if any of the tube creation processes fail.
+  #
+  # @return [void]
+  def create_racked_tubes(racked_tubes, new_tube_rack)
+    # iterate through the rack attributes to create each tube
+    racked_tubes.each { |tube_attributes| create_racked_tube(tube_attributes, new_tube_rack) }
+  end
+
+  def create_racked_tube(tube_attributes, new_tube_rack)
+    tube = create_tube(tube_attributes)
+    link_tube_to_rack(tube, new_tube_rack, tube_attributes[:tube_position])
+  end
+
+  # Ensures that the provided tube barcode is unique.
+  #
+  # This method checks if a tube barcode already exists in the database.
+  # If the barcode is found, it raises a StandardError indicating that the barcode is already in use.
+  # If the barcode is not found, the method simply returns, allowing the process to continue.
+  #
+  # @param [String] tube_barcode The barcode of the tube to be checked for uniqueness.
+  #
+  # @raise [StandardError] if the tube barcode is already in use.
+  #
+  # @return [void]
+  def ensure_unique_tube_barcode(tube_barcode)
+    existing_tube_barcode_record = Barcode.includes(:asset).find_by(asset_id: tube_barcode)
+    return if existing_tube_barcode_record.nil?
+
+    error_message = "The tube barcode '#{tube_barcode}' is already in use, cannot continue."
+    raise StandardError, error_message
+  end
+
+  # Checks the format of the provided tube barcode.
+  #
+  # This method verifies that the provided tube barcode matches a recognized format.
+  # It first checks if the barcode format is recognized. If not, it raises a StandardError.
+  # Then, it checks if the barcode format is of the expected 'fluidx_barcode' type.
+  # If the barcode format is not 'fluidx_barcode', it raises a StandardError.
+  #
+  # @param [String] tube_barcode The barcode of the tube to be checked.
+  #
+  # @raise [StandardError] if the barcode format is not recognized or if it is not of the
+  # expected 'fluidx_barcode' type.
+  #
+  # @return [void]
+  def check_tube_barcode_format(tube_barcode)
+    barcode_format = Barcode.matching_barcode_format(tube_barcode)
+
+    # barcode format should be recognised
+    if barcode_format.nil?
+      error_message = "The tube barcode '#{tube_barcode}' is not a recognised format."
+      raise StandardError, error_message
+    end
```

Comment:
Some code duplicated between here and `create_new_barcode` method.

Reply:
one is for tube and the other for tuberack.
wondering if these should all be in the models extending from a common labware model method.